envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataTest.kt

```diff
@@ -181,4 +181,71 @@ class NodeMetadataTest {
         assertThat(dependency.service).isEqualTo("service-1")
         assertThat(dependency.settings.handleInternalRedirect).isEqualTo(true)
     }
+
+    @Test
+    fun `should accept incoming settings with custom healthCheck path`() {
+        // given
+        val proto = proxySettingsProto(
+            incomingSettings = true,
+            path = "/path",
+            healthCheckPath = "/status/ping"
+        )
+        val incoming = proto.structValue?.fieldsMap?.get("incoming").toIncoming()
+
+        // expects
+        assertThat(incoming.healthCheck.clusterName).isEqualTo("local_service")
+        assertThat(incoming.healthCheck.path).isEqualTo("/status/ping")
+        assertThat(incoming.healthCheck.hasHealthCheck()).isTrue()
+    }
+
+    @Test
+    fun `should set empty healthCheck path for incoming settings when path is empty`() {
+        // given
+        val proto = proxySettingsProto(
+            incomingSettings = true,
+            path = "/path"
```

Comment:
Path is not empty here, healthCheckPath is. Title a little bit misleading / or variable name.